### PR TITLE
Feature: Memory Banks (Windows Only)

### PR DESCRIPTION
**Pull request by KingRial**
When collecting HW informations about memory, it's also very usefull to know the physical memory banks and their specs

Quick example:
image

### CODE REVIEW

File: memory.go [hunk 1]
```diff
@@ -11,9 +11,19 @@ import (
 	"math"
 )
 
+type MemoryBank struct {
+	Name         string `json:"name"`
```
Comment (jaypipes) [inline memory.go]: Let's go ahead and remove the `Name` field. I don't think it adds anything over the `Label` and `Location` fields.

File: memory.go [hunk 2]
```diff
@@ -11,9 +11,19 @@ import (
 	"math"
 )
 
+type MemoryBank struct {
```
Comment (jaypipes) [inline memory.go]: Let's call this `MemoryModule`, to align with industry terminology:

https://en.wikipedia.org/wiki/Memory_module

I know CMI/WMI uses the term "bank", but generally these things are referred to as "modules" in my experience :)

File: memory_windows.go [hunk 3]
```diff
@@ -21,12 +21,49 @@ type win32OperatingSystem struct {
 	TotalVisibleMemorySize uint64
 }
 
+const wmqlPhysicalMemor = "SELECT BankLabel, Capacity, DataWidth, Description, DeviceLocator, Manufacturer, Model, Name, PartNumber, PositionInRow, SerialNumber, Speed, Tag, TotalWidth FROM Win32_PhysicalMemory"
```
Comment (jaypipes) [inline memory_windows.go]: Missing "y" at the end of the above variable name :)

File: memory_windows.go [hunk 4]
```diff
@@ -21,12 +21,49 @@ type win32OperatingSystem struct {
...
+	DataWidth     uint16
+	Description   string
+	DeviceLocator string
+	Manufacturer  string
+	Model         string
+	Name          string
+	PartNumber    string
+	PositionInRow uint32
+	SerialNumber  string
+	Speed         uint32
```
Comment (jaypipes) [inline memory_windows.go]: In a future PR (not this one), I'd be interested in adding a `Speed` field to the `MemoryModule` struct. Need to have a conversation about what the appropriate data type is, though, so let's leave that for a future commit.

File: memory_windows.go [hunk 5]
```diff
@@ -21,12 +21,49 @@ type win32OperatingSystem struct {
...
+	for _, description := range win32MemDescriptions {
+		info.Banks = append(info.Banks, &MemoryBank{
+			Name:         description.Description,
+			Label:        description.BankLabel,
+			Location:     description.DeviceLocator,
+			SerialNumber: description.SerialNumber,
+			SizeBytes:    int64(description.Capacity),
+			Vendor:       description.Manufacturer,
+		})
+		//totalPhysicalBytes += description.Capacity
```
Comment (jaypipes) [inline memory_windows.go]: Remove above comment please.

**Comment by KingRial** [comment]
As soon as I'll get back from my travels, I'll follow the suggestions

**Comment by jaypipes** [comment]
@KingRial in trying to rebase this, I totally goofed it up and ended up `git cherry-picking` the original commit from you into a new PR here: https://github.com/jaypipes/ghw/pull/172

Closing this one out! :)

**Pull request closed without merging**